### PR TITLE
fix: add response upgrade checks and logs

### PR DESCRIPTION
**Pull request by RomanAyupov89**
https://team-1602965683919.atlassian.net/browse/NLJ-2397 #in-progress

### CODE REVIEW

File: src/proxy_handler.cr [hunk 1]
```diff
@@ -20,14 +20,23 @@ class Mitm::ProxyHandler
...
+      response = context.response
+      response.upgrade do |io|
+        counter += 1
+        Log.trace { "Connect upgrade counter: #{counter}" }
+        if io.closed?
+          Log.trace { "Connect upgrade io closed counter: #{counter}" }
+          next
+        end
+        peek = io.peek
+        next if peek.nil?
```
Comment (alonchin) [inline src/proxy_handler.cr:36]: what about adding a trace log in this block?

File: src/proxy_handler.cr [hunk 2]
```diff
@@ -20,14 +20,23 @@ class Mitm::ProxyHandler
...
+        magic = peek.[1]?
         next if magic.nil?
```
Comment (alonchin) [inline src/proxy_handler.cr:38]: what about adding a trace log in this block?

File: src/proxy_handler.cr (hunk 1 above)
Comment (RomanAyupov89) [reply src/proxy_handler.cr]: fixed

File: src/proxy_handler.cr (hunk 2 above)
Comment (RomanAyupov89) [reply src/proxy_handler.cr]: fixed

**Review by alonchin — APPROVED** [review]

**Pull request merged**